### PR TITLE
Rework Permission

### PR DESCRIPTION
**Pull request by Lulalaby**

### CODE REVIEW

File: DisCatSharp/Enums/Permission.cs [hunk 1]
```diff
@@ -90,209 +90,323 @@ public enum Permissions : long
...
+		VoiceChannelPermissions = Connect | Speak | Video | UseActivities | PrioritySpeaker |
+			MuteMembers | DeafenMembers | MoveMembers,
+
+		/// <summary>
+		/// Includes all stage channel permissions.
+		/// </summary>
+		[PermissionString("Stage Channel Permissions")]
+		StageChannelPermissions = RequestToSpeak,
+
+
```
Comment (Lulalaby) [inline DisCatSharp/Enums/Permission.cs:139]:
```suggestion

```

File: DisCatSharp/Enums/Permission.cs [hunk 2]
```diff
@@ -90,209 +90,321 @@ public enum Permissions : long
...
 		/// </summary>
-		[PermissionString("No permissions")]
+		[PermissionString("No Permissions")]
 		None = 0x0000000000000000,
 
 		/// <summary>
 		/// Indicates all permissions are granted
 		/// </summary>
-		[PermissionString("All permissions")]
+		[PermissionString("All Permissions")]
```
Comment (Saalvage) [inline DisCatSharp/Enums/Permission.cs:99]: I'm not exactly sure where these appear, but what's the rationale for using title case, as opposed to sentence case as is used in the official DC docs?

File: DisCatSharp/Enums/Permission.cs [hunk 3]
```diff
@@ -90,209 +90,321 @@ public enum Permissions : long
 		/// <summary>
 		/// Indicates no permissions given.
 		/// </summary>
-		[PermissionString("No permissions")]
+		[PermissionString("No Permissions")]
 		None = 0x0000000000000000,
 
 		/// <summary>
 		/// Indicates all permissions are granted
```
Comment (Saalvage) [inline DisCatSharp/Enums/Permission.cs:97]: Missing dot.

File: DisCatSharp/Enums/Permission.cs [hunk 4]
```diff
@@ -90,209 +90,321 @@ public enum Permissions : long
...
+		/// <summary>
+		/// Includes all advanced permissions.
+		/// </summary>
+		[PermissionString("Advanced Permissions")]
+		AdvancedPermissions = Administrator,
+
+		/// <summary>
+		/// Allows members to invite new people to this server.
+		/// </summary>
+		[PermissionString("Create Instant Invite")]
```
Comment (Saalvage) [inline DisCatSharp/Enums/Permission.cs:153]: Why don't we just mirror the Discord docs descriptions here?

File: DisCatSharp/Enums/Permission.cs [hunk 5]
```diff
@@ -304,40 +416,73 @@ public enum Permissions : long
...
+		[PermissionString("Send Messages in Threads and Posts")]
+		SendMessagesInThreadsAndPosts = 0x0000004000000000,
+
+		/// <summary>
+		/// Allows members to use Activities in this server.
 		/// </summary>
-		[PermissionString("Start Embedded Activities")]
-		StartEmbeddedActivities = 0x0000008000000000,
+		[PermissionString("Use Activities")]
+		UseActivities = 0x0000008000000000,
```
Comment (Saalvage) [inline DisCatSharp/Enums/Permission.cs:459]: Full name should be "UseEmbeddedActivities" according to DC docs?

**Review by Saalvage — CHANGES_REQUESTED** [review]

File: DisCatSharp/Enums/Permission.cs (hunk 2 above)
Comment (Lulalaby) [reply DisCatSharp/Enums/Permission.cs:99]: Client-Side

File: DisCatSharp/Enums/Permission.cs [hunk 6]
```diff
@@ -90,209 +90,321 @@ public enum Permissions : long
...
+		/// Includes all advanced permissions.
+		/// </summary>
+		[PermissionString("Advanced Permissions")]
+		AdvancedPermissions = Administrator,
+
+		/// <summary>
+		/// Allows members to invite new people to this server.
+		/// </summary>
+		[PermissionString("Create Instant Invite")]
 		CreateInstantInvite = 0x0000000000000001,
```
[the remark this answers is not in the document]
Comment (Lulalaby) [reply DisCatSharp/Enums/Permission.cs:154]: true

**Pull request closed without merging**